metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -279,16 +289,35 @@ object BuildServerConnection {
       }
     }
 
-    setupServer().map { connection =>
-      new BuildServerConnection(
-        setupServer,
-        connection,
-        languageClient,
-        reconnectNotification,
-        config,
-        workspace
-      )
-    }
+    setupServer()
+      .map { connection =>
+        new BuildServerConnection(
+          setupServer,
+          connection,
+          languageClient,
+          reconnectNotification,
+          config,
+          workspace
+        )
+      }
+      .recoverWith { case NonFatal(e) =>
+        scribe.warn("Could not connect to build server")
```

Comment:
if we catch all `NonFatal` I would probably log this 

```suggestion
        scribe.warn("Could not connect to build server", e)
```

(I think scribe allows for passing `e` as a second parameter?)

Reply:
Actually I removed the message, this message will be reported in `setupServer()`, so this would just be duplicate.